modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -374,45 +374,49 @@ const FrontendDataSet = ({
 		}
 	}, [wrapperRef]);
 
-	function refreshData(successNotification) {
-		setDataLoading(true);
-
-		return requestData()
-			.then(({data}) => {
-				if (successNotification?.showSuccessNotification) {
-					openToast({
-						message:
-							successNotification.message ||
-							Liferay.Language.get('table-data-updated'),
-						type: 'success',
-					});
-				}
+	const refreshData = useCallback(
```

Comment:
Addition of `useCallback` here is to get rid of `// eslint-disable-next-line react-hooks/exhaustive-deps` on [this line](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L553). It's not an ideal solution, but does the job for now. Ideal solution would be to refactor code and have a single method for fetching data, instead of somewhat-duplicating and confusing `loadData`, `refreshData`, `reloadData` and `requestData`.